### DISCUSSION TITLE
2FA Not Always Triggered

### DISCUSSION THREAD

**Discussion by Andrew-J-Larson** [community/community#57717] [Mobile]
I know that since then there's been an additional prompt added to the GitHub 2FA on PC to wait for you to open the app on mobile... But I just wanted to say that at least on the Android app how I was working around this was I was turning the battery usage mode to unrestricted. So that it wouldn't have problems of not prompting. Not sure if anyone on the dev team knew about this, but if they don't, it would be useful in not requiring an additional prompt to open the mobile app.

**Comment by iskae** [comment]
Hi @TheAlienDrew 👋 

Thanks so much for reaching out ❤️ This is definitely interesting, and we would love to learn a bit more on this 🙇 

Can you elaborate a bit more on this by answering the following questions?

- Which device are you using and which Android version is running on the device?
- Are you setting the battery saver mode or low power mode manually? 

Thanks in advance!

**Reply by Andrew-J-Larson** [reply]
Previously, I was using a Samsung Galaxy S21 5G, but I have since gotten a new phone and now I'm using a Google Pixel 7.

And yes, I would manually set the setting, but I know there is a programmatic way to get permissions from the user to set this, as I've seen in other apps that require similar 2FA push features.

**Reply by Andrew-J-Larson** [reply]
I think also "Pause app activity if unused" is something that also needs to be turned off, and I don't remember if there were any apps that did more than just redirect to the app setting for that for the user to switch off, so that might also be something to keep in consideration.
